extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/ExcludedTypeBuildItem.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.arc.deployment;
+
+import io.quarkus.builder.item.MultiBuildItem;
+
+/**
+ * This build item is used to specify types to be excluded from discovery.
+ * 
+ * <p>
+ * An element value can be:
+ * <ul>
+ * <li>a fully qualified class name, i.e. {@code org.acme.Foo}</li>
+ * <li>a simple class name as defined by {@link Class#getSimpleName()}, i.e. {@code Foo}</li>
+ * <li>a package name with suffix {@code .*}, i.e. {@code org.acme.*}, matches a package</li>
+ * <li>a package name with suffix {@code .**}, i.e. {@code org.acme.**}, matches a package that starts with the value</li>
+ * </ul>
+ * If any element value matches a discovered type then the type is excluded from discovery, i.e. no beans and observer
+ * methods are created from this type.
+ */
+public final class ExcludedTypeBuildItem extends MultiBuildItem {
```

Comment:
I think the name should be more self explanatory. We have no idea it's excluded from what. I'm pretty sure @mkouba  will come with a good name.

Reply:
Yeah, I used the same name as the config. I honestly don't care what it's called, but the same name as the config doesn't seem too far-fetched, does it?

Reply:
So… yes/no?

Reply:
It depends on the question ;-). I'd say yes for the `ExcludedTypeBuildItem`.

Reply:
Well, then let's merge this ;)